### ISSUE TITLE
Add a tileset for land use

### ISSUE THREAD

**Issue by amcewen** [Liverpool-UK/15-minute-city#13]
Related to #5, it would be good to have a tileset that showed different land usage so you could see if your 15-minute-isochrone included green space, somewhere to shop, places to work (both in an office and somewhere more industrial), somewhere to live, amenities (schools, libraries, etc.)

**Comment by amcewen** [comment]
Have been looking at building our own tiles for that, using [cyclOSM](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/docs/INSTALL.md) as a starting point (partly because it's bike-focused, partly because it's documented...).

1. Followed [the cyclosm install guide](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/docs/INSTALL.md).
    * Installed `postgresql-12` (as 9.6 isn't readily available on this Ubuntu install)
    * Skipped installing `python-gdal` as it isn't available and *looks* like it's part of `gdal-bin` now anyway.
    * Restricted NASA download to `-a -4:50:-1:55` as we don't need the whole world...  Actually that failed, so I skipped the whole hillshading section
    * Used England for the contour generation http://download.geofabrik.de/europe/great-britain/england.poly
1. Set up a new user because npm doesn't like installing things as root (should've done this ages ago really)
1. Installed `kosmtik` from the [cyclosm contributing guide](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/CONTRIBUTING.md#rendering-locally--developping-the-style)
    1. Installed `nvm` because the kosmtik doesn't work with new versions of NodeJS: `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
    1. Logged out and back in.  Installed Node 6 LTS: `nvm install 6.17.1`
    1. `npm -g install kosmtik`
    1. `~/.nvm/versions/node/v6.17.1/bin/kosmtik plugins --install kosmtik-fetch-remote` (just running `kosmtik` seemed to find a global one I'd installed while discovering I needed to use nvm...)
    1. `sudo apt install python-dev libmapnik-dev`
1. Created a default `localconfig.json`, copying it from [the example in the install docs](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/docs/INSTALL.md#kosmtik)
1. Run kosmtik: `~/.nvm/versions/node/v6.17.1/bin/kosmtik serve project.mml`
1. Now you can open http://localhost:6789/ but at present I don't seem to have any tiles to see...

1. Had to download the shp file zips - https://osmdata.openstreetmap.de/download/land-polygons-split-3857.zip and https://osmdata.openstreetmap.de/download/simplified-land-polygons-complete-3857.zip, unzip them, and update the `project.mml` to point to them
1. Because I downloaded a collection of `pbf` files for the OSM data (for Merseyside, Lancashire, Cheshire and Greater Manchester) I needed to reload them into the database (otherwise I only got the last one I loaded):
    * `osm2pgsql -c -s -G --hstore -d osm ./merseyside-latest.osm.pbf`
    * `osm2pgsql -a -s -G --hstore -d osm ./cheshire-latest.osm.pbf`
    * `osm2pgsql -a -s -G --hstore -d osm ./lancashire-latest.osm.pbf`
    * `osm2pgsql -a -s -G --hstore -d osm ./greater-manchester-latest.osm.pbf`

And now I have a tile server!  Now I just need to work out how to style things differently...

![image](https://user-images.githubusercontent.com/108468/88489939-b3adb480-cf8f-11ea-9b1c-f66e3438db7c.png)

**Comment by amcewen** [comment]
To get this running again I needed to do the following, while `ssh`ed into the 15minutecity.mcqn.com server:
  * `cd cyclosm-cartocss-style`
  * `nvm install 6.17.1`
  * `~/.nvm/versions/node/v6.17.1/bin/kosmtik serve project.mml`
Plus I needed to create an ssh tunnel to port 6789 on localhost to be able to access the server.

**Comment by amcewen** [comment]
https://blog.datawrapper.de/beautifulcolors/ has some good tips on choosing sets of colours to use

**Comment by amcewen** [comment]
What different land uses are we interested in?
 * Water - lakes, rivers, streams, seas, etc.
 * Parks, fields, etc.
 * Residential - houses, flats
 * Retail
 * Amenities - libraries, council offices, leisure centres/gyms, galleries and museums, hospitals, places of worship
 * Blue-collar workplaces - warehouses, factories, mechanics
 * White-collar workplaces - offices

Kosmtik uses [CartoCSS](https://github.com/mapbox/carto) to describe how to render the maps.  The [Mapbox CartoCSS manual](https://github.com/mapbox/mapbox-studio-classic/blob/mb-pages/docs/studio-classic-manual/04-classic-manual-cartocss.md) is a decent starting point to find out more.

I expect it'll be useful to use the CartoCSS variables to define the different landuse colours.
```
 @landuse-water = #73b6e6;
 @landuse-parks = #73e6b6;
 @landuse-amenities = #e67373;
 ...
```

The colours used in CyclOSM are all defined in `palette.mss`.

Given the amount of different types of data, etc. to render, it makes more sense to modify the CyclOSM styles than it does to start from scratch.  So, we have forked that into https://github.com/Liverpool-UK/landuse-cartocss-style and cloned it and set things up on the server to allow us to render and show that tileset.

Next we'll be able to start making serious changes to the carto-css to start designing our map.

**Comment by amcewen** [comment]
Of course, there's already a [landuse OSM project](https://osmlanduse.org/).  We should check it out to see if we're reinventing the wheel :-)

**Comment by jodischneider** [comment]
Definitely looks relevant and high quality - the question is how/where to get the data. 

So I wrote Prof. Dr. Alexander Zipf whose team is behind the osmlanduse.org project to see if there is any open data available. Their research project involves combining the highest-resolution [Copernicus data](https://spacedata.copernicus.eu/fr/dataset-details?articleId=394198) with OpenStreetMap, and using deep learning [per their blog post](http://k1z.blog.uni-heidelberg.de/2020/10/06/osmlanduse-european-union-validation-effort/). Currently they're part of an EU project seeking citizen scientists [to help validate the deep learning results](https://landsense.eu/Mapathon). Will pass on what I find!
